openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,690 @@
+            if field_error:
+                errors = True
+                continue
+            replacement_collection = None
+            if template_field.replacement_collection:
+                replacement_collection = (
+                    template_field.replacement_collection.collection
+                )
+
+            for replacement in replacements:
+                structured_field = self.make_structured(template_field, replacement)
+                if structured_field not in model:
+                    self.errors.append(
+                        f"{collection}/{model['id']}/{template_field.own_field_name}: Missing {structured_field} since it is given as a replacement"
+                    )
+                    errors = True
+
+                if replacement_collection:
+                    try:
+                        as_id = int(replacement)
+                    except (TypeError, ValueError):
+                        self.errors.append(
+                            f"{collection}/{model['id']}/{template_field.own_field_name}: Replacement {replacement} is not an integer"
+                        )
+                    if not self.find_model(replacement_collection, as_id):
+                        self.errors.append(
+                            f"{collection}/{model['id']}/{template_field.own_field_name}: Replacement {replacement} does not exist as a model of collection {replacement_collection}"
+                        )
+
+            for field in model.keys():
+                if self.is_structured_field(field):
+                    try:
+                        _template_field, _replacement = self.to_template_field(
+                            collection, field
+                        )
+                        if (
+                            template_field.own_field_name == _template_field
+                            and _replacement not in model[template_field.own_field_name]
+                        ):
+                            self.errors.append(
+                                f"{collection}/{model['id']}/{field}: Invalid structured field. Missing replacement {_replacement} in {template_field.own_field_name}"
+                            )
+                            errors = True
+                    except CheckException as e:
+                        self.errors.append(
+                            f"{collection}/{model['id']}/{field} error: " + str(e)
+                        )
+                        errors = True
+
+        return errors
+
+    def check_types(self, model: Dict[str, Any], collection: str) -> None:
+        for field in model.keys():
+            if self.is_template_field(field):
+                continue
+
+            field_type = self.get_type_from_collection(field, collection)
+            enum = self.get_enum_from_collection_field(field, collection)
+
+            checker: Optional[Callable[..., bool]] = None
+            if isinstance(field_type, CharField):
```

Comment:
Why did you remove the mapping? I think it is cleaner to list it there and only access it here (the mapping would be of type `Dict[Field, Callable]` now, of course)

Reply:
I got problems with the type checking and tried Dict[Field, Callable]. It did not work.

Reply:
Doesn't mypy tell you what type it actually is? If you can't figure it out, you can still put `Any` or `# type: ignore` there, but restructuring the code because you can't figure out the correct type is usually not the correct approach.

Reply:
Use now Any, now it is okay. Use the checker_map again.